docs/Bridging_Assets.md

```diff
@@ -0,0 +1,158 @@
+
+### Canonical Token Bridge Implementation
+
+The two main bridging contracts are EthERC20Bridge.sol (L1 side) and ArbTokenBridge.sol (L2 side). The system offers two options for creating an L2 pairing for a token contract: the standard ERC20 option and the "custom token" option. Custom tokens should only be used if you have a good reason to.
+
+#### Standard Arb-ERC20 Bridging
+
+Any ERC20 token on Ethereum can be bridged onto Arbitrum "the standard way" by simply calling `EthERC20Bridge.deposit` with the token's L1 address:
+
+```sol
+    /**
+     * @notice Deposit standard or custom ERC20 token. If L2 side hasn't been deployed yet, includes name/symbol/decimals data for initial L2 deploy.
+     * @param erc20 L1 address of ERC20
+     * @param destination account to be credited with the tokens in the L2 (can be the user's L2 account or a contract)
+     * @param amount Token Amount
+     * @param maxSubmissionCost Max gas deducted from user's L2 balance to cover base submission fee
+     * @param maxGas Max gas deducted from user's L2 balance to cover L2 execution
+     * @param gasPriceBid Gas price for L2 execution
+     * @param callHookData optional data for external call upon minting
+     * @return ticket ID used to redeem the retryable transaction in the L2
+     */
+    function deposit(
+        address erc20,
+        address destination,
+        uint256 amount,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid,
+        bytes calldata callHookData
+    ) external payable override returns (uint256)
+```
+
+If no paired-L2 has yet been deployed (i.e., of this is the first deposit), a [StandardArbERC20](https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/StandardArbERC20.sol) will be deployed on Arbitrum at a deterministically generated address. StandardArbErc20 implements the ERC20 standard with additional with additional bridging-related methods (i.e., `bridgeMint, withdraw`, etc.) as well as extensions to improve UX (ERC1363 & ERC2612). The L1 contracts name, symbol, and decimals will also be pushed directly into the new StandardArbERC20.
+
+To withdraw from a a standard ERC20 from Arbitrum , call `SomeStandardArbERC20.withdraw`
+
+```sol
+    /**
+     * @notice Initiates a token withdrawal
+     * @param account destination address
+     * @param amount amount of tokens withdrawn
+     */
+    function withdraw(address account, uint256 amount) external override {
+```
+
+This initiates the withdrawal process (i.e., tokens can later be claimed via the Outbox.)
```

Comment:
Should we mention you can't withdraw a token that was natively deployed to the L2?

Reply:
good call, will cover in its own section